src/app/view/perspectives/post-it/post-it-perspective.component.ts

```diff
@@ -373,32 +360,27 @@ export class PostItPerspectiveComponent implements OnInit, OnDestroy {
     return Math.max(1, Math.floor(layoutWidth / postItWidth));
   }
 
-  private getCollectionRoles(postIt: PostItDocumentModel): string[] {
+  public getCollectionRoles(postIt: PostItDocumentModel): string[] {
     return this.collectionRoles && this.collectionRoles[postIt.document.collectionId] || [];
   }
 
-  private sortByOrder(item: any, element: HTMLElement): number {
-    return Number(element.getAttribute('order'));
-  }
-
   public trackByDocument(index: number, postIt: PostItDocumentModel): number {
-    return HashCodeGenerator.hashString(postIt.document.id || postIt.document.correlationId);
+    return postIt ? postIt.hash() : Number.MIN_SAFE_INTEGER;
   }
 
   public ngOnDestroy(): void {
-    if (this.navigationHelper) {
-      this.navigationHelper.destroy();
-    }
-
-    if (this.infiniteScroll) {
-      this.infiniteScroll.destroy();
-    }
+    this.unsubscribeIfPossible(this.deletionHelper);
+    this.unsubscribeIfPossible(this.navigationHelper);
+    this.unsubscribeIfPossible(this.configHelper);
+    this.unsubscribeIfPossible(this.infiniteScroll);
+    this.unsubscribeIfPossible(this.collectionsSubscription);
+    this.subscriptions.forEach(subscription => subscription.unsubscribe());
```

Comment:
Maybe it would be worth using just two subscriptions here - one for post-its and one for all other things.